diff --git a/src/core/net/ip6.cpp b/src/core/net/ip6.cpp
index b510fb12c..b7bf7526e 100644
--- a/src/core/net/ip6.cpp
+++ b/src/core/net/ip6.cpp
@@ -1047,6 +1047,15 @@ Error Ip6::ProcessReceiveCallback(Message &          aMessage,
             break;
         }
 
+#if OPENTHREAD_CONFIG_TCP_ENABLE
+        // Do not pass TCP message to avoid dual processing from both openthread and POSIX tcp stacks
+        case kProtoTcp:
+        {
+            error = kErrorNoRoute;
+            goto exit;
+        }
+#endif
+
         default:
             break;
         }
