diff --git a/src/posix/platform/openthread-core-posix-config.h b/src/posix/platform/openthread-core-posix-config.h
index 6708751f2..ba6079bce 100644
--- a/src/posix/platform/openthread-core-posix-config.h
+++ b/src/posix/platform/openthread-core-posix-config.h
@@ -184,7 +184,7 @@
  *
  */
 #ifndef OPENTHREAD_CONFIG_MLE_MAX_CHILDREN
-#define OPENTHREAD_CONFIG_MLE_MAX_CHILDREN 64
+#define OPENTHREAD_CONFIG_MLE_MAX_CHILDREN 128
 #endif
 
 /**
