From d5655ab9649c5e754edd0a22a9504f752c7f0750 Mon Sep 17 00:00:00 2001
From: LiJunmeng <junmeng.li_1@nxp.com>
Date: Wed, 7 Aug 2024 18:35:04 +0530
Subject: [PATCH 1/2] Add imx8mm evk iw612 otbr dtb support

Signed-off-by: LiJunmeng <junmeng.li_1@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 3deec3664e38..8db629e8b13b 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -97,6 +97,8 @@ dtb-$(CONFIG_ARCH_MXC) += imx8mm-ddr4-evk.dtb imx8mm-ddr4-evk-rm67191.dtb imx8mm
 			  imx8mm-ddr4-evk-rm67199.dtb \
 			  imx8mm-ddr4-evk-rm67191-cmd-ram.dtb \
 			  imx8mm-ddr4-evk-rm67199-cmd-ram.dtb
+dtb-$(CONFIG_ARCH_MXC) += imx8mm-evk-uart3-bt-ffu-gpio-irq.dtb
+dtb-$(CONFIG_ARCH_MXC) += imx8mm-evk-uart3-bt-3M-ffu-gpio-irq.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mm-emcon-avari.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mm-emtop-baseboard.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mm-icore-mx8mm-ctouch2.dtb
-- 
2.25.1

