From d81e6ec26269ee6157411903f5fe3e94e9989198 Mon Sep 17 00:00:00 2001
From: Haoran Wang <elven.wang@nxp.com>
Date: Fri, 18 Nov 2022 13:26:33 +0000
Subject: [PATCH 3/4] Install pskc for otbr-agent credentials generation

Signed-off-by: Haoran Wang <elven.wang@nxp.com>
---
 tools/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index 5e8451afdb..3832e98ea6 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -46,6 +46,8 @@ target_link_libraries(steering-data PRIVATE
     mbedtls
 )
 
+install(TARGETS pskc  DESTINATION sbin)
+
 if ($ENV{REFERENCE_DEVICE})
     add_subdirectory(reference_device)
 endif()
-- 
2.34.1

